src/renderer/components/+preferences/application.tsx

```diff
@@ -56,7 +62,7 @@ export const Application = observer(() => {
 
   const [customUrl, setCustomUrl] = React.useState(userStore.extensionRegistryUrl.customUrl || "");
   const [shell, setShell] = React.useState(userStore.shell || "");
-  const extensionSettings = AppPreferenceRegistry.getInstance().getItems().filter((preference) => preference.showInPreferencesTab === "application");
+  const extensionSettings = appPreferenceItems.get().filter((preference) => preference.showInPreferencesTab === "application");
```

Comment:
To go even further, there could be an injectable whose type is

```
IComputedValue<Map<string, RegisteredAppPreference[]>>
```

which does the splitting once

Reply:
I don't follow...

Reply:
We currently have one injectable: `appPreferenceItemsInjectable`. But we could change its type to from `IComputedValue<RegisteredAppPreference[]>` to `IComputedValue<Map<string, RegisteredAppPreference[]>>` and then have an injectable for each of the pages, say called `telemetryPreferenceItemsInjectables`. That way when we come to test each component, we only need to override the single injectable and the filtering is moved out of the component.

Reply:
Sounds good. But that may be better done in another PR? After https://github.com/lensapp/lens/pull/4651 is merged, which is waiting on this PR.

have to resolve conflicts first...

Reply:
Okay sounds good